### DISCUSSION TITLE
Youtube Download + Gradio + Whisper + Facebook M2M_100 to add subtitles

### DISCUSSION THREAD

**Discussion by R4ZZ3** [openai/whisper#283] [Show and tell]
https://huggingface.co/spaces/Finnish-NLP/Whisper-ASR-youtube-subtitles

Created a demo with Whisper + Facebook M2M_100 to add subtitles to videos spoken in any of the languages supported by Whisper
1. Download youtube video with a given url and then you can watch it in the gradio demo video component
2. Run Whisper ASR on the video and translate to different languages
3. Burn translation to original video and watch in  another video component

**Comment by TheOddPirate** [comment]
Hope u fond a en-no model, would be Just what i need

**Reply by R4ZZ3** [reply]
I changed it to work with 100 languages using facebook M2M100_418M
@n0tOdd

**Reply by TheOddPirate** [reply]
Thank you, now i have everything i need to get subtitles for all my movies and series =)

**Reply by R4ZZ3** [reply]
For longer videos I suggest you to use this locally or modify the code to your needs. The demo space might not work for really long videos unless you split processing into smaller chunks. I am working on adding faster c++ based model to this based on https://github.com/ggerganov/whisper.cpp and then add some splitting functionality to split longer videos to smaller chunks

**Reply by R4ZZ3** [reply]
@n0tOdd @toprak 
New version out. Now with ggerganovs c++ implementation + deepL for fast and accurate translations in 26 languages
https://huggingface.co/spaces/RASMUS/Whisper-youtube-crosslingual-subtitles

**Comment by ghost** [comment]
does it work on yt playlists to?
